lib/vrl/stdlib/src/type_def.rs

```diff
@@ -0,0 +1,129 @@
+use ::value::kind::Unknown;
+use vrl::prelude::*;
+
+use vrl::prelude::TypeDef as VrlTypeDef;
+
+fn type_def(type_def: &VrlTypeDef) -> Resolved {
+    let mut tree = BTreeMap::new();
+    insert_if_true(&mut tree, "fallible", type_def.is_fallible());
+    insert_kind(&mut tree, type_def.kind(), true);
```

Comment:
I'd prefer it if we moved this logic into the `value` crate (either as the `Debug` impl, or as a custom method), which then would (1) allow us to use this in `dbg!()` statements as well (something I use frequently), and it would avoid having to make the two collection fields public in the `Kind` type.

Reply:
I'll move it over as a custom method.